docs/community/maintainers-guide/releasing.rst

```diff
@@ -0,0 +1,224 @@
+
+When you need to create a new release for a maintained previous release,
+you will have to do the extra step of backporting the changes to the
+maintenance branch.
+
+After the procedure described above, you will have to:
+
+1. Checkout the ``maint`` branch and create a new release branch:
+    - ``git checkout maint-<major>.<minor>``
+    - ``git checkout -b rel-v<major>.<minor>.<patch>``
+2. Cherry-pick the commits that you need to backport, resolve any conflict.
+   (yes, cherry-pick :))
+3. Run tests.
+4. Createa new commit with updated ``changes`` file and bumped version,
+   as described above.
+5. Issue a pull request against the **maintenance branch** (
+   ``maint-<major>.<minor>``).
+6. If Travis fails:
+    - Fix issue and ensure head commit is the release commit (i.e. rebase if
+      necessary).
+7. Merge, tag and push release:
+    - ``git merge --ff-only rel-v<major>.<minor>.<patch>``
+    - ``git tag v<major>.<minor>.<patch>``
+    - ``git push inveniosoftware maint-<major>.<minor> v<major>.<minor>.<patch>``
+
+Invenio release
+---------------
+
+The pre-requirement necessary to release Invenio is that all the features in
+the various Invenio modules needed for the release have been merged and
+released. Then:
+
+1. Update the ``setup.py``:
+    - review all modules lower and upper bounds and adjust them as needed.
+2. Documentation:
+    - review documentation and make sure new features or breaking changes are
+      documented, to help users when upgrading.
+3. Prepare release notes (`see example <https://github.com/inveniosoftware/invenio/commit/f4d0aa5ac78d76228fe86754eeb3bbfe81a1854f>`_):
+    - In ``docs/releases/``, copy an existing patch-level or minor
+      release notes (e.g. ``docs/releases/v3.1.2.rst``).
+    - Edit release notes.
+    - Include the new release notes into ``docs/releases/index.rst``.
+    - Check the "Maintenance policy", e.g. is the version correct?
+      (`example <https://github.com/inveniosoftware/invenio/commit/edb863d2f5228fb158c090a69c2db7c3385b6ba3>`_).
+4. Create a release commit (message pattern:
+   ``release: v<major>.<minor>.<patch>``).
+5. Create tag and push release:
+    - ``git tag v<major>.<minor>.<patch>``
+    - ``git push inveniosoftware master v<major>.<minor>.<patch>``
+6. Regularly check ``Travis`` and ``PyPI`` to ensure the release went through.
+
+Manual releases
+---------------
+
+When the process of releasing fails for some reasons, you might want to
+manually publish the new version of a package.
+
+**PyPI**
+
+You can manually reproduce the publishing process done by ``Travis`` by doing:
+
+1. Activate your virtualenv for the package that you want to release.
+2. Generate the different distributions:
+    - ``python setup.py compile_catalog sdist bdist_wheel``
+3. Install the tool to upload releases to PyPI:
+    - ``pip install twine wheel``
+4. Publish:
+    - ``twine upload --repository <repo-name> dist/*``. The command will
```

Comment:
Thinking of newcomers to the role: maybe we could point out that they can test the release in https://test.pypi.org/ (see [docs](https://twine.readthedocs.io/en/latest/#using-twine)).

Reply:
- I think that the `<repo-name>` placeholder in `twine upload --repository <repo-name> dist/*` can be confused with the Invenio module repository. 
- We can add a link to the architects: https://github.com/orgs/inveniosoftware/teams/architects
```console
4. Publish:
    - ``twine upload dist/*``. The command will ask for username and password. `Invenio architects <https://github.com/orgs/inveniosoftware/teams/architects>`_ should have the credentials.
.. note::
   If it is your first time releasing in PyPI, or you are not sure if the release is correct, you can test it in https://test.pypi.org/ see the `documentation 
<https://twine.readthedocs.io/en/latest/#using-twine>`_.
```